homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +94,160 @@ async def async_setup_entry(
         return f"Seventeentrack Package: {name}"
 
     @property
     def native_value(self) -> StateType:
         """Return the state."""
-        return self._state
-
-    async def async_update(self) -> None:
-        """Update the sensor."""
-        await self._data.async_update()
-
-        if not self.available:
-            # Entity cannot be removed while its being added
-            async_call_later(self.hass, 1, self._remove)
-            return
-
-        package = self._data.packages.get(self._tracking_number, None)
-
-        # If the user has elected to not see delivered packages and one gets
-        # delivered, post a notification:
-        if package.status == VALUE_DELIVERED and not self._data.show_delivered:
-            self._notify_delivered()
-            # Entity cannot be removed while its being added
-            async_call_later(self.hass, 1, self._remove)
-            return
-
-        self._attr_extra_state_attributes.update(
-            {
-                ATTR_INFO_TEXT: package.info_text,
-                ATTR_TIMESTAMP: package.timestamp,
-                ATTR_LOCATION: package.location,
-            }
-        )
-        self._state = package.status
-        self._friendly_name = package.friendly_name
-
-    async def _remove(self, *_):
-        """Remove entity itself."""
-        await self.async_remove(force_remove=True)
-
-        reg = er.async_get(self.hass)
-        entity_id = reg.async_get_entity_id(
-            "sensor",
-            "seventeentrack",
-            UNIQUE_ID_TEMPLATE.format(self._data.account_id, self._tracking_number),
-        )
-        if entity_id:
-            reg.async_remove(entity_id)
-
-    def _notify_delivered(self):
-        """Notify when package is delivered."""
-        _LOGGER.info("Package delivered: %s", self._tracking_number)
-
-        identification = (
-            self._friendly_name if self._friendly_name else self._tracking_number
+        package_data = self.coordinator.data.current_packages.get(
+            self._tracking_number, {}
         )
-        message = NOTIFICATION_DELIVERED_MESSAGE.format(
-            identification, self._tracking_number
-        )
-        title = NOTIFICATION_DELIVERED_TITLE.format(identification)
-        notification_id = NOTIFICATION_DELIVERED_TITLE.format(self._tracking_number)
+        return package_data["package"].status
 
-        persistent_notification.create(
-            self.hass, message, title=title, notification_id=notification_id
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes."""
+        package_data = self.coordinator.data.current_packages.get(
+            self._tracking_number, {}
         )
+        return package_data["extra"]
 
 
-class SeventeenTrackData:
-    """Define a data handler for 17track.net."""
+def remove_entity(hass: HomeAssistant, account_id: str, tracking_number: str) -> bool:
```

Comment:
Can we maybe just pass the list of packages it should remove so it can do a for loop here?

Reply:
done